### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'state')

### ISSUE THREAD

**Issue by Fonger** [grpc/grpc-node#2347]
### Problem description

After we upgrade `@grpc/grpc-js` from 1.7.3 to 1.8.7, we start to see server crash frequently with this error:

```
TypeError: Cannot read properties of undefined (reading 'state')
    at RetryingCall.sendMessageWithContext (/workspace/node_modules/@grpc/grpc-js/build/src/retrying-call.js:542:22)
    at /workspace/node_modules/@grpc/grpc-js/build/src/resolving-call.js:106:19
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

This map to this line 597 `call.state === 'ACTIVE'`
https://github.com/grpc/grpc-node/blob/d209a34d6f95e4db8c36bbfb074f6e8729950d41/packages/grpc-js/src/retrying-call.ts#L593-L598

### Reproduction steps

Sorry I can't give a precise step to reproduce for now, however maybe it's possible to find the issue by code review?

I suspect it's because `this.commitedCallIndex` is `-1` because `this.underlyingCalls` is an empty array.

https://github.com/grpc/grpc-node/blob/d209a34d6f95e4db8c36bbfb074f6e8729950d41/packages/grpc-js/src/retrying-call.ts#L267-L275

Or committedCallIndex is null because `this.state` is `COMMITED` or `COMPLETED` but we force nullish! here.
https://github.com/grpc/grpc-node/blob/d209a34d6f95e4db8c36bbfb074f6e8729950d41/packages/grpc-js/src/retrying-call.ts#L595

Or one of element of underlyingCalls maybe undefined so we should use optional chaining like this one here: (I think this one is unlikely)
https://github.com/grpc/grpc-node/blob/d209a34d6f95e4db8c36bbfb074f6e8729950d41/packages/grpc-js/src/retrying-call.ts#L624

The only dependency we use `@grpc/grpc-js` is opentelemetry `@opentelemetry/exporter-trace-otlp-grpc` 0.34.0. 

### Environment
 - OS name, version and architecture: Alpine Linux v3.17
 - Node version: v18.14.0
 - Node installation method: docker node:18-alpine
 - If applicable, compiler version:
 - Package name and version: @grpc/grpc-js@1.8.7

**Comment by murgatroid99** [comment]
I think I fixed the issue in version 1.8.8. Please update and try it out.

**Comment by Fonger** [comment]
> I think I fixed the issue in version 1.8.8. Please update and try it out.

Thanks! I can confirm this is fixed in 1.8.8.
